include/wx/gtk/slider.h

```diff
@@ -9,6 +9,8 @@
 #ifndef _WX_GTK_SLIDER_H_
 #define _WX_GTK_SLIDER_H_
 
+#include "wx/gtk/private/wrapgtk.h"
```

Comment:
As Travis CI build failures show, you can't include this private header from a public one. This means that you can't use `GTK_CHECK_VERSION()` here neither, you need to always declare the virtual functions but implement them differently depending on the version check.

Reply:
Okay, this was a little awkward because it also means that I can't use native GTK+ structures (GtkPositionType in this case) as a member. I suppose it makes sense to decouple the toolkit specific implementation. 

As a result, I have added code to `wxSlider::SetTick()` to convert the style to a tick position 'on the fly'. This might not be quite optimal, but it is relatively simple and understandable. I am hoping that it is safe to use `GetWindowStyle()` here.

There are a couple of ways I can see of using `GTK_CHECK_VERSION()`:
1. Wrap the internal implementation of each function so that the functions compile as empty for older toolkit versions
2. Implement two versions of each function (one of which is empty), and use `GTK_CHECK_VERSION()` to determine which version is compiled.
The second option would result in less conditional directives as all functions could be wrapped in a single branch, but I have chosen the first simply because it is consistent with the way it has been done in `wxListBox::GetTopItem()`

Reply:
I think the current version is just fine, i.e. I agree both with converting the style to GTK enum on the fly and with using version checks inside the functions -- although, to be honest, I'd be fine with the other implementation too.

The only thing I might change would be to add `wxUnusedVar(param)` to avoid warnings about unused parameters with GTK < 2.16, but this is hardly critical, of course.

So if you don't have any objections, I'm going to merge this PR soon. Thanks again!

Reply:
> The only thing I might change would be to add wxUnusedVar(param) to avoid warnings about unused parameters with GTK < 2.16, but this is hardly critical, of course.

Good spot, it's always best to have a clean compile. I've added them now, but cannot test as I don't have a GTK < 2.16 build environment.

I also tidied up the spaces around the 'if' statements.

I don't have any objections, please go ahead and merge if you think it's ready. Thanks!